polkadot/node/network/collator-protocol/src/collator_side/metrics.rs

```diff
@@ -230,12 +249,12 @@ impl metrics::Metrics for Metrics {
 pub(crate) const MAX_BACKING_DELAY: BlockNumber = 3;
 // Paras availability period. In practice, candidates time out in exceptional situations.
 pub(crate) const MAX_AVAILABILITY_DELAY: BlockNumber = 10;
+// Finality delay for included blocks.
+pub(crate) const MAX_FINALITY_DELAY: BlockNumber = MAX_FINALITY_LAG as BlockNumber;
 
-// Collations are kept in the tracker, until they are included or expired
+// Collations are kept in the tracker, until they are finalized or expired
 #[derive(Default)]
 pub(crate) struct CollationTracker {
-	// Keep track of collation expiration block number.
-	expire: HashMap<BlockNumber, HashSet<Hash>>,
```

Comment:
There was an issue with expiration: we didn't reschedule expiration after blocks moved to the next stage. I rewrote it without using an additional structure.